scanomatic/io/app_config.py

```diff
@@ -249,12 +249,11 @@ def _generate_admin_uuid(self) -> Optional[str]:
         return val
 
     def save_current_settings(self) -> None:
-        if self.validate():
-            dump(
-                self._settings,
-                self._paths.config_main_app,
-                merge=True,
-            )
+        dump(
+            self._settings,
+            self._paths.config_main_app,
+            merge=True,
+        )
```

Comment:
Why no `validate`?

Reply:
please re-add validation, though it should probably not be `self.validate` but from the `validate.py`

Reply:
I'm confused :confused: . I was sure this change was done by @joakimmoller but it looks like it's mine. I'll reintroduce it

Reply:
I think it might have been me that broke this when updating models but any model derived from `generics.model.Model` (or what the path is should be possible to validate with `validate.py` without any fuss.

Reply:
Looking at this more closely I'm not sure it's wrong.

`self.validate()` was removed previously so if validation should occur it should def be from the `validate.py`

But the only place `save_current_settings()` is called is here where the settings are already validated before saving, https://github.com/Scan-o-Matic/scanomatic-standalone/blob/a95d1082989c2620eac64ce3757ae063f1bacbe9/scanomatic/ui_server/settings_api.py#L42

Importing the `validate()` function also causes a big mess of circular imports.

Reply:
I see, in that case we don't need to validate it twice. Perhaps just add a docstring to the `save_current_settings` that it should be validated beforehand and also perhaps it would be nicer to not save `self._settings` but `self.application_settings` since that is what is validated.

Reply:
Added docstring and changed `self._settings -> self.application_settings`